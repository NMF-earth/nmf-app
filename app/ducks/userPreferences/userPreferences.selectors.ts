import { pathOr } from "ramda";
import { ElectricityEnum } from "carbon-footprint";

import { namespace } from "./userPreferences.slice";

const getAcceptedTermsOfUseVersion = (state) =>
  pathOr(0, [namespace, "acceptedTermsOfUseVersion"], state);

const getLocation = (state) =>
  pathOr(ElectricityEnum.world, [namespace, "location"], state);

export default {
  getAcceptedTermsOfUseVersion,
  getLocation,
};
