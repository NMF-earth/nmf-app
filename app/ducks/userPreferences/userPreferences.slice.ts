import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { ElectricityType } from "carbon-footprint";

import { currentLanguage } from "utils";

const initialState = {
  acceptedTermsOfUseVersion: 0,
  activatedNotifications: false,
  location: ElectricityType.world,
  language: currentLanguage,
  timesStarted: 0,
  useMetricUnits: true,
};

const userPreferences = createSlice({
  name: "userPreferences",
  initialState,
  reducers: {
    acceptTermsOfUse(state, action: PayloadAction<number>) {
      state.acceptedTermsOfUseVersion = action.payload;
    },
    toggleNotifications(state, action: PayloadAction<boolean>) {
      state.activatedNotifications = action.payload;
    },
    updateLocation(state, action: PayloadAction<ElectricityType>) {
      state.location = action.payload;
    },
    changeLanguage(state, action: PayloadAction<string>) {
      state.language = action.payload;
    },
    incrementTimesStarted(state) {
      state.timesStarted += 1;
    },
    toggleUnits(state, action: PayloadAction<boolean>) {
      state.useMetricUnits = action.payload;
    },
  },
});

const {
  acceptTermsOfUse,
  updateLocation,
  toggleNotifications,
  changeLanguage,
  incrementTimesStarted,
  toggleUnits,
} = userPreferences.actions;

export const actions = {
  acceptTermsOfUse,
  updateLocation,
  toggleNotifications,
  changeLanguage,
  incrementTimesStarted,
  toggleUnits,
};

export const namespace = userPreferences.name;

export const reducer = userPreferences.reducer;
