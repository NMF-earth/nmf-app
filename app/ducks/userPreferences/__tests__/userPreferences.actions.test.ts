import { ElectricityEnum } from "carbon-footprint";

import userPreferences from "../";

describe("userPreferences actions should", () => {
  it("export expected actions", () =>
    expect(userPreferences.actions).toMatchSnapshot());

  it("be able to update to a new term of use version", () => {
    const acceptedTermsOfUseVersion = 2;

    const expectedAction = {
      type: userPreferences.actions.acceptTermsOfUse.toString(),
      payload: acceptedTermsOfUseVersion,
    };
    expect(
      userPreferences.actions.acceptTermsOfUse(acceptedTermsOfUseVersion)
    ).toEqual(expectedAction);
  });

  it("be able to update to update user location", () => {
    const expectedAction = {
      type: userPreferences.actions.updateLocation.toString(),
      payload: ElectricityEnum.france,
    };
    expect(
      userPreferences.actions.updateLocation(ElectricityEnum.france)
    ).toEqual(expectedAction);
  });
});
