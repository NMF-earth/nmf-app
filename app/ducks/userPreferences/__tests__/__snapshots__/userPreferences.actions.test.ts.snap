// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userPreferences actions should export expected actions 1`] = `
{
  "acceptTermsOfUse": [Function],
  "changeLanguage": [Function],
  "incrementTimesStarted": [Function],
  "toggleNotifications": [Function],
  "toggleUnits": [Function],
  "updateLocation": [Function],
}
`;
